src/bin/bat/config.rs

```diff
@@ -15,6 +16,46 @@ pub fn config_file() -> PathBuf {
         .unwrap_or_else(|| PROJECT_DIRS.config_dir().join("config"))
 }
 
+pub fn generate_config_file() {
+    let config_file = config_file();
+    if config_file.exists() {
+        println!("A config file already exists at: {}", config_file.to_string_lossy());
+
+        print!("Overwrite? (y/n): ");
+        let _ = io::stdout().flush();
+        let mut decision = String::new();
+        io::stdin().read_line(&mut decision).expect("Failed to read input");
+
+        if !decision.trim().eq_ignore_ascii_case("Y") {
+            return;
+        }
+    } else {
+        let config_dir = config_file.parent().unwrap();
+        if !config_dir.exists() {
+            fs::create_dir(config_dir).expect("Unable to create config directory");
+        }
+    }
+
+    let default_config = "# Specify desired theme (e.g. \"TwoDark\")
```

Comment:
To prevent having to escape all quotes in this default config, you can use a raw string:
``` rust
let default_config = r#"…
…"#;
```